### PR TITLE
Feature/1053/update promocode

### PR DESCRIPTION
**Pull request by afanasieva-anastasiia**
## Description

Add possibility to updating promocode on backend part. #1053 
Add test for updating promocode method and fix tests for other mutations.
Remove unnecessary code.
Add pagination for promocodes on backend part. #1085 
Create validation for promocodes on backend part. #1062 

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date front-end and admin part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

**Review by maarekua — DISMISSED** [review]

File: modules/promo-code/promo-code.permissions.js [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+const { hasRoles, inputDataValidation } = require('../../utils/rules');
 const {
   roles: { ADMIN, SUPERADMIN },
 } = require('../../consts');
+const {
+  PROMOCODE_FIELDS: { PROMOCODE },
+} = require('../../consts/promocode-fields');
 
 const promoCodeQuery = {
+  getPromoCodeById: allow,
```
Comment (AndriyLymych) [inline modules/promo-code/promo-code.permissions.js:15]: everybody can do it? even logged out users?

File: modules/promo-code/promo-code.service.js [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
 const {
   STATUS_CODES: { NOT_FOUND },
 } = require('../../consts/status-codes');
 const {
   PROMOCODE_NOT_FOUND,
-} = require('../../error-messages/promocode.messages');
+} = require('../../error-messages/promocode-messages');
 
 class PromoCodeService {
   async getAllPromoCodes() {
```
Comment (AndriyLymych) [inline modules/promo-code/promo-code.service.js:11]: pagination for this?

File: tests/promo-code/promo-code.mutation.test.js [hunk 3]
```diff
@@ -1,32 +1,54 @@
...
-  it('should add add to database', async () => {
+  it('should add to database', async () => {
     promoCodes = await addPromoCode(newPromoCode, operations);
 
     expect(promoCodes).toHaveProperty('code', newPromoCode.code);
-    expect(promoCodes.dateFrom).toEqual('2021-12-26');
+    expect(promoCodes.dateFrom).toEqual(newPromoCode.dateFrom);
     expect(promoCodes).toHaveProperty('discount', newPromoCode.discount);
-    expect(promoCodes.discount).toEqual(12);
+    expect(promoCodes.discount).toEqual(newPromoCode.discount);
```
Comment (AndriyLymych) [inline tests/promo-code/promo-code.mutation.test.js:29]: why do you need a lot of expects for this? just one common

File: tests/promo-code/promo-code.mutation.test.js [hunk 4]
```diff
@@ -1,32 +1,54 @@
...
   });
 
   it('delete promo-code', async () => {
-    promoCodeId = promoCodes._id;
     const res = await deletePromoCode(promoCodeId, operations);
     promoCodes = res.data.deletePromoCode;
+
     expect(promoCodes).toHaveProperty('code', newPromoCode.code);
-    expect(promoCodes.dateFrom).toEqual('2021-12-26');
+    expect(promoCodes.dateFrom).toEqual(newPromoCode.dateFrom);
```
Comment (AndriyLymych) [inline tests/promo-code/promo-code.mutation.test.js:51]: same

File: modules/promo-code/promo-code.service.js (hunk 2 above)
Comment (afanasieva-anastasiia) [reply modules/promo-code/promo-code.service.js:11]: pagination will be implemented in another pull request

Comment (AndriyLymych) [reply modules/promo-code/promo-code.service.js:11]: why? its just few lines

Comment (AndriyLymych) [reply modules/promo-code/promo-code.service.js:11]: ok, create new PR for this case

File: tests/promo-code/promo-code.mutation.test.js (hunk 4 above)
Comment (afanasieva-anastasiia) [reply tests/promo-code/promo-code.mutation.test.js]: Fixed

File: tests/promo-code/promo-code.mutation.test.js (hunk 3 above)
Comment (afanasieva-anastasiia) [reply tests/promo-code/promo-code.mutation.test.js]: Fixed

File: modules/promo-code/promo-code.permissions.js (hunk 1 above)
Comment (afanasieva-anastasiia) [reply modules/promo-code/promo-code.permissions.js]: Improve permission

**Review by bohdanst1914 — APPROVED** [review]

**Review by bohdanst1914 — DISMISSED** [review]

File: modules/promo-code/promo-code.service.js [hunk 5]
```diff
@@ -1,28 +1,31 @@
...
     const items = await PromoCode.find().exec();
+    const count = PromoCode.find().countDocuments();
```
Comment (SirAndrii) [inline modules/promo-code/promo-code.service.js:14]: do we need an additional request to DB if we have already got the object?

Comment (afanasieva-anastasiia) [reply modules/promo-code/promo-code.service.js:14]: Fixed this

File: tests/promo-code/promo-code.mutation.test.js [hunk 6]
```diff
@@ -1,32 +1,49 @@
...
 let promoCodes;
 let promoCodeId;
 let operations;
 
-describe('PromoCodes queries', () => {
-  beforeAll(async () => {
+describe('PromoCodes mutations', () => {
+  beforeEach(async () => {
     operations = await setupApp();
+    promoCodes = await addPromoCode(newPromoCode, operations);
```
Comment (VasylLymych) [inline tests/promo-code/promo-code.mutation.test.js:19]: Not sure that we need add promocode before each test, same as call setupApp function. I think it' more proper to use beforeAll here.

Comment (afanasieva-anastasiia) [reply tests/promo-code/promo-code.mutation.test.js]: I agree. Improved this

**Review by maarekua — APPROVED** [review]

**Review by LubomyrKashuba — APPROVED** [review]

**Pull request merged**